app/helpers/ledger.js

```diff
@@ -0,0 +1,175 @@
+import { wallet } from "wallet-preload-shim";
+import { hexToBytes, strHashToRaw } from "helpers";
+import * as secp256k1 from "secp256k1";
+import { default as blake } from "blake-hash";
+import * as bs58 from "bs58";
+import toBuffer from "typedarray-to-buffer";
+import { getTxFromInputs } from "../actions/TransactionActions";
+
+export function addressPath(branch, index) {
+  const prefix = "44'/42'/0'/";
+  const i = (index || 0).toString();
+  const b = (branch || 0).toString();
+  return prefix + b + "/" + i;
+}
+
+export function pubkeyToHDPubkey(pubkey, chainCode, isTestnet) {
+  const pk = secp256k1.publicKeyConvert(hexToBytes(pubkey), true); // from uncompressed to compressed
+  const cc = hexToBytes(chainCode);
+  let hdPublicKeyID = hexToBytes("02fda926"); // dpub
+  if (isTestnet) {
+    hdPublicKeyID = hexToBytes("043587d1"); // tpub
+  }
+  const parentFP = hexToBytes("00000000"); // not true but we dont know the fingerprint
```

Comment:
Not %100 sure if this is ok to do. I wonder when the parent fingerprint is ever needed?